### PR TITLE
Added CMake to project so it can be cross-compiled

### PR DESCRIPTION
**Pull request by mwallner**

### CODE REVIEW

**Comment by mwallner** [comment]
Hey @Gfilinic - I took the liberty of creating a PR for your changes.
Please let us know when you think it's fit to be included in upstream, I'd love to be able to X-compile builds on GitHub actions.

**Comment by Gfilinic** [comment]
Hello!

Thank you for showing interest! I was actually thinking of just PR because i think it is useful. But i got distracted with other work... I think everything is set up alright, the only thing that could be better is to remove the build scipt, as it is unnecessary. Other than that I've found no issues with the building process, as i've used it to build via Yocto for custom Raspi OS. 

Let me know if there is anything more that I can do. 

P.S.  I am new to this kind of contribution work, so bare with me.... I think i approved the merge request.

**Review by aurelian17 — APPROVED** [review]
A very good improvement.

**Comment by mstroh76** [comment]
I’m uncertain about how to handle ‘pins’ and ‘wiringpid’:
 - ‘pins,’ there’s an issue with the 40-pin GPIO, and it requires LaTeX and dvipdf.
 - ‘wiringpid,’ it’s outdated, unused (no information found online), and may pose a security risk.

Perhaps we should consider dropping support for both. At the very least, I believe they shouldn’t be included in the main ‘CMakeLists.txt’ file, and we should remove the ‘install(’ part."

**Comment by mstroh76** [comment]
it was already done  #187 #169 , maybe @Lucino772 wants to review it too

**Comment by Lucino772** [comment]
Just took a quick look and it seems good to me. I think the only differences are that I extracted the version from the `version.h` file and I also used [CPack ](https://cmake.org/cmake/help/latest/module/CPack.html) to generate a **.deb** package.

**Comment by joggee-fr** [comment]
Nice work @Gfilinic.
@mstroh76, I see this PR has been merged in a dedicated branch named `cmake`. Is it planned for the branch to be merged in `master` in the future?

File: devLib/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+set(HEADERS
+    ds1302.h maxdetect.h piNes.h
+    gertboard.h piFace.h
+    lcd128x64.h lcd.h
+    scrollPhatFont.h scrollPhat.h
+    piGlow.h
+)
+
+# Compiler flags
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wformat=2 -Winline -pipe -fPIC")
```
Comment (joggee-fr) [inline devLib/CMakeLists.txt:25]: I think `-fPIC` is not really needed here as position independent code seems to be the default for shared libraries with CMake: https://cmake.org/cmake/help/v3.30/prop_tgt/POSITION_INDEPENDENT_CODE.html#prop_tgt:POSITION_INDEPENDENT_CODE.

File: devLib/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# CMakeLists.txt for devLib
+
+cmake_minimum_required(VERSION 3.10)
+project(devLib)
+
+# Source files
+set(SRC
+    ds1302.c maxdetect.c piNes.c
```
Comment (joggee-fr) [inline devLib/CMakeLists.txt:8]: May use `file(GLOB)` to build the list: https://cmake.org/cmake/help/v3.30/command/file.html#glob.

File: gpio/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+# Add the executable target
+add_executable(gpio ${SRC})
+
+# Find the required libraries
+find_library(WIRINGPI_LIB wiringPi)
+find_library(WIRINGPI_DEV_LIB wiringPiDev)
+
+# Link the required libraries
+target_link_libraries(gpio ${WIRINGPI_LIB} ${WIRINGPI_DEV_LIB} pthread rt m crypt)
```
Comment (joggee-fr) [inline gpio/CMakeLists.txt:23]: May be good to find `libcrypt` on the system first e.g. using `pkg-config`:

```
find_package(PkgConfig REQUIRED)
pkg_check_modules(libcrypt REQUIRED)
target_link_libraries(gpio [...]  PkgConfig::libcrypt)
```

**Pull request merged**